android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -847,14 +1094,85 @@ private WritableArray getVideoTrackInfo() {
                 videoTrack.putInt("height",format.height == Format.NO_VALUE ? 0 : format.height);
                 videoTrack.putInt("bitrate", format.bitrate == Format.NO_VALUE ? 0 : format.bitrate);
                 videoTrack.putString("codecs", format.codecs != null ? format.codecs : "");
-                videoTrack.putString("trackId",
-                        format.id == null ? String.valueOf(trackIndex) : format.id);
-                videoTracks.pushMap(videoTrack);
+                videoTrack.putString("trackId", format.id == null ? String.valueOf(trackIndex) : format.id);
+                if (isFormatSupported(format)) {
+                    videoTracks.pushMap(videoTrack);
+                }
             }
         }
+
         return videoTracks;
     }
 
+    private WritableArray getVideoTrackInfoFromManifest(Timeline timeline) {
+        return this.getVideoTrackInfoFromManifest(timeline, 0);
+    }
+
+    private WritableArray getVideoTrackInfoFromManifest(Timeline timelineRef, int retryCount) {
```

Comment:
I don't understand why you need a retry strategy to get Video Tracks ?
Is it related to loadManifest which can be long ?

Reply:
It can take long and I think that the manifest request can fail for whatever reason so we added a retry to attempt another load if it was an intermittent network issue etc.

Reply:
Let's add a comment noting this.